src/main/java/uk/gov/hmcts/reform/sscs/config/TribunalsHearingsEventQueueListenerConfig.java

```diff
@@ -0,0 +1,103 @@
+
+@Slf4j
+@Component
+@ConditionalOnProperty("flags.tribunals-to-hearings-api.enabled")
+public class TribunalsHearingsEventQueueListenerConfig {
+
+    private final HearingsService hearingsService;
+
+    @Value("${azure.service-bus.tribunals-to-hearings-api.connectionString}")
+    private String connectionString;
+    @Value("${azure.service-bus.tribunals-to-hearings-api.queueName}")
+    private String queueName;
+
+    @Value("${azure.service-bus.tribunals-to-hearings-api.retryTimeout}")
+    private Long retryTimeout;
+    @Value("${azure.service-bus.tribunals-to-hearings-api.retryDelay}")
+    private Long retryDelay;
+    @Value("${azure.service-bus.tribunals-to-hearings-api.maxRetries}")
+    private Integer maxRetries;
+
+    @Autowired
+    public TribunalsHearingsEventQueueListenerConfig(HearingsService hearingsService) {
+        this.hearingsService = hearingsService;
+    }
+
+    @EventListener(ApplicationReadyEvent.class)
+    @SuppressWarnings("PMD.CloseResource")
+    public void tribunalsHearingsEventProcessorClient() {
+        ServiceBusSessionReceiverClient  receiverClient = new ServiceBusClientBuilder()
+            .retryOptions(retryOptions())
+            .connectionString(connectionString)
+            .sessionReceiver()
+            .queueName(queueName)
+            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+            .disableAutoComplete()
+            .buildClient();
+
+        log.info("Tribunals hearings event queue receiver starting.");
+
+        CompletableFuture.runAsync(() -> {
+            while (true) {
+                processMessage(receiverClient);
+            }
+        });
+    }
+
+    private void processMessage(ServiceBusSessionReceiverClient  receiverClient) {
+
+        try (ServiceBusReceiverClient receiver = receiverClient.acceptNextSession()) {
+            receiver.receiveMessages(1).forEach(message -> {
+                try {
+                    HearingRequest hearingRequest = message.getBody().toObject(HearingRequest.class);
+                    String caseId = hearingRequest.getCcdCaseId();
+                    HearingState event = hearingRequest.getHearingState();
+                    log.info("Attempting to process hearing event {} from hearings event queue for case ID {}",
+                        event, caseId);
+
+                    hearingsService.processHearingRequest(hearingRequest);
+
+                    receiver.complete(message);
+                    log.info("Hearing event {} for case ID {} successfully processed", event, caseId);
+                } catch (Exception ex) {
+                    log.error("An exception occurred whilst processing hearing event for case ID {}."
+                        + " Abandoning message", message.getMessageId(), ex);
+                    receiver.abandon(message);
+                }
+            });
+        } catch (Exception ex) {
+            log.error("Error occurred while closing the session", ex);
+        }
```

Comment:
Catching Exception can be dangerous.  Better to catch the expected exception types so others feed up to the top.
E.g. if a NullPointerException is thrown, catching that here could have consequences further up stream.
Are we sure we want to catch it so generically in these cases?